lib/IPCPipeSend.cpp

```diff
@@ -0,0 +1,91 @@
+/** @file IPCPipeSend.cpp
+ *  @author Bao Vu (bao.vu@unikie.com)
+ *  @brief
+ */
+
+#include "IPCPipeSend.h"
+
+IPCPipeSend::~IPCPipeSend()
+{
+	if (this->pd > 0)
+		close(this->pd);
+	unlink(this->opts.server_name.c_str());
+}
+
+void IPCPipeSend::init()
+{
+	// remove old pipe (if any)
+	remove(this->opts.server_name.c_str());
+	// create a fifo
+	if (mkfifo(this->opts.server_name.c_str(), 0660) != 0)
+		throw std::runtime_error("ERROR: " + this->opts.server_name + ": " + strerror(errno));
+	// open the pipe with O_NONBLOCK to get EAGAIN
+	this->pd = open(this->opts.server_name.c_str(), O_RDWR | O_NONBLOCK);
```

Comment:
will this pd used for reading?

Reply:
No, but it's allow me to open pipe before reader start.